src/main/java/org/gridsuite/ds/server/service/DynamicSimulationService.java

```diff
@@ -28,48 +29,63 @@
  */
 @Service
 public class DynamicSimulationService {
-
-    private final ResultRepository resultRepository;
-
-    @Autowired
-    private StreamBridge publishRun;
-
     private static final String CATEGORY_BROKER_OUTPUT = DynamicSimulationService.class.getName() + ".output-broker-messages";
-
     private static final Logger LOGGER = LoggerFactory.getLogger(CATEGORY_BROKER_OUTPUT);
+    private final ResultRepository resultRepository;
+    private final NotificationService notificationService;
+    private final DynamicMappingClient dynamicMappingClient;
+    private final ParametersService parametersService;
 
-    public DynamicSimulationService(ResultRepository resultRepository) {
+    public DynamicSimulationService(ResultRepository resultRepository, NotificationService notificationService, DynamicMappingClient dynamicMappingClient, ParametersService parametersService) {
         this.resultRepository = Objects.requireNonNull(resultRepository);
+        this.notificationService = Objects.requireNonNull(notificationService);
+        this.dynamicMappingClient = Objects.requireNonNull(dynamicMappingClient);
+        this.parametersService = Objects.requireNonNull(parametersService);
     }
 
-    public Mono<UUID> runAndSaveResult(UUID networkUuid, String variantId, int startTime, int stopTime, FilePart dynamicModel) {
-
-        Mono<byte[]> fileBytes;
-        fileBytes = dynamicModel.content().collectList().flatMap(all -> Mono.fromCallable(() ->
-                StreamUtils.copyToByteArray(new DefaultDataBufferFactory().join(all).asInputStream())));
-
-        return fileBytes.flatMap(bytes -> {
-            DynamicSimulationRunContext runContext = new DynamicSimulationRunContext(networkUuid, variantId, startTime, stopTime, bytes);
-            // update status to running status and store the dynamicModel file
-            return insertStatus(DynamicSimulationStatus.RUNNING.name())
-                    .flatMap(resultEntity ->
-                            Mono.fromRunnable(() -> {
-                                Message<String> message = new DynamicSimulationResultContext(resultEntity.getId(), runContext).toMessage();
-                                sendRunMessage(message);
-                            })
-                                    .thenReturn(resultEntity.getId())
-                    );
-        });
+    public Mono<UUID> runAndSaveResult(String receiver, UUID networkUuid, String variantId, int startTime, int stopTime, String mappingName) {
+
+        return dynamicMappingClient.createFromMapping(mappingName) // get script and parameters file from dynamic mapping server
```

Comment:
To see later, but I'm not sure it is relevant to continue using spring reactor for services implementation.

Reply:
Noted.. to discuss